### PR TITLE
[WIP] Add support for Gogs.

### PR DESCRIPTION
**Pull request by tosone**
**What this PR does / why we need it**:

Add support for Gogs CI/CD.

/cc @zhujian7

**Release note**:

```release-note
Add support for Gogs CI/CD
```

### CODE REVIEW

File: pkg/server/biz/scm/gogs/gogs.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	}
+	for _, b := range gogsBranches {
+		branches = append(branches, b.Name)
+	}
+	return
+}
+
+// ListTags list all of repo's tags
+func (g *Gogs) ListTags(repoName string) (tags []string, err error) {
+	err = cerr.ErrorNotImplemented.Error("get tag list")
```
Comment (zhujian7) [inline pkg/server/biz/scm/gogs/gogs.go]: Why not return error directly?

**Comment by zhujian7** [comment]
@tosone Please declare the supported version scope of Gogs in [user-guide](https://github.com/caicloud/cyclone/blob/master/docs/user_guide.md#integration-center)

File: pkg/server/biz/scm/gogs/parsewebhook.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			return
+		}
+		eventData = &scm.EventData{
+			Type: scm.PushEventType,
+			Repo: pushPayload.Repo.FullName,
+			Ref:  pushPayload.Ref,
+		}
+		return
+	default:
+		log.Warningln("Skip unsupported Github event")
```
Comment (zhujian7) [inline pkg/server/biz/scm/gogs/parsewebhook.go]: `Github` -> `Gogs` ?

File: pkg/server/biz/scm/gogs/gogs.go [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+// GetToken get Gogs's token
+func (g *Gogs) GetToken() (token string, err error) {
+	token, err = genTokenByBasicAuth(g.scmCfg.Server, g.scmCfg.User, g.scmCfg.Password)
+	return
+}
+
+// genTokenByBasicAuth generate token by username and password
+func genTokenByBasicAuth(server, user, password string) (token string, err error) {
+	if len(server) == 0 {
+		err = fmt.Errorf("Gitea server is missing")
```
[the remark this answers is not in the document]
Comment (tosone) [reply pkg/server/biz/scm/gogs/gogs.go]: Done

File: pkg/server/biz/scm/gogs/parsewebhook.go (hunk 2 above)
Comment (tosone) [reply pkg/server/biz/scm/gogs/parsewebhook.go]: Done

**Comment by supereagle** [comment]
@tosone Any updates?

**Comment by supereagle** [comment]
@tosone I will close this PR, feel free to reopen it if necessary.

/close

**Pull request closed without merging**